src/rgw/services/svc_bi_rados.cc

```diff
@@ -207,10 +207,10 @@ int RGWSI_BucketIndex_RADOS::open_bucket_index(const DoutPrefixProvider *dpp,
     return ret;
   }
 
-  get_bucket_index_objects(bucket_oid_base, idx_layout.layout.normal.num_shards,
+  get_bucket_index_objects(bucket_oid_base, rgw::num_shards(idx_layout.layout.normal),
                            idx_layout.gen, bucket_objs, shard_id);
   if (bucket_instance_ids) {
-    get_bucket_instance_ids(bucket_info, idx_layout.layout.normal.num_shards,
+    get_bucket_instance_ids(bucket_info, rgw::num_shards(idx_layout.layout.normal),
```

Comment:
..but won't this break access to existing buckets that had num_shards=0?

Reply:
yeah I am still testing this.